### ISSUE TITLE
Signature for `Enumerable#min|max_by` doesn't support a block that returns `Array`

### ISSUE THREAD

**Issue by marianosimone** [sorbet/sorbet#1231]
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Aa%20%3D%20%5B1%2C2%2C3%5D%0Ab%20%3D%20a.max_by%20%7B%7Ci%7C%20%5Bi%2C%20-i%5D%7D)

```ruby
a = [1,2,3]
b = a.max_by {|i| [i, -i]}
```

#### Observed output

```
Returning value that does not conform to block result type https://srb.help/7005
     3 |b = a.max_by {|i| [i, -i]}
            ^^^^^^^^^^^^^^^^^^^^^^
  Expected Comparable
  Got [Integer, Integer] originating from:
    editor.rb:3:
     3 |b = a.max_by {|i| [i, -i]}
                          ^^^^^^^
Errors: 1
```

#### Expected behavior
It should be allowed to return an `Array` from the block, as that's the way to sort by multiple criteria.

Given that `Array`s are not `Comparable` (see [here](https://bugs.ruby-lang.org/issues/5574)), maybe we could at least support returning `Comparable` OR `Array[Comparable]`

- - -

**Comment by aisamanra** [comment]
I think this is absolutely something we want to support. I think it's interesting that we're not consistent, because I believe that `sort_by` does effectively the same thing as `{min,max}_by` (i.e. it takes a block that returns a thing which supports `<=>`) but for that signature we assume the block returns `BasicObject` instead, which allows it to work over arrays but also over lots of other types that will fail at runtime. It might be best to `{max,min,sort}_by` consistent all at once.

**Comment by DarkDimius** [comment]
Fixed in #1235
